### PR TITLE
feat(observe,server): support filtering executions by multiple agentflow IDs

### PR DESCRIPTION
**Pull request by jocelynlin-wd**
## Summary

- **Server**: `GET /executions` accepts the `agentflowId` query in any HTTP idiom — single (`?agentflowId=a`), comma-separated (`?agentflowId=a,b`), repeated keys (`?agentflowId=a&agentflowId=b`), or mixed — all flatten into a single `string[]`. Service uses one `IN (:...agentflowIds)` clause for both single and multi cases. Single-id callers continue to work identically.
- **SDK (`@flowiseai/observe`)**: `ExecutionsViewerProps` and `ExecutionFilters` take `agentflowIds: string[]`. API client serializes to the canonical comma-joined form for wire-format stability regardless of axios `paramsSerializer` overrides.
- **Example app**: `VITE_FLOW_IDS` env var (one or comma-separated UUIDs); demo header surfaces multi-agentflow scope.

Motivation: an app on the consuming side can have multiple agentflows, so the executions viewer needs to scope across all of them in a single request — single-id filtering forces lossy client-side merging or leaks rows from other apps.

## Test plan

- [x] `tsc --noEmit` clean for `flowise`, `@flowiseai/observe`, and the example app
- [x] New `controllers/executions/index.test.ts` — 9 tests covering all four wire shapes + whitespace/empty edge cases
- [x] API client tests added for comma-joined serialization + empty-array omission (13/13 pass)
- [x] Manual: in the example app with `VITE_FLOW_IDS=<id1>,<id2>` set, the executions list shows merged rows from both agentflows
- [x] Manual: omitting `VITE_FLOW_IDS` shows the unscoped global list
- [x] Manual: with a single id (`VITE_FLOW_IDS=<id>`), behavior matches the previous single-flow scoped view

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Recordings:

https://github.com/user-attachments/assets/431d71d0-65e2-41f7-9a03-dae7cdc83e41

https://github.com/user-attachments/assets/be5599f8-2451-40aa-999c-5ea3d0cce1dd

https://github.com/user-attachments/assets/f827d0ed-8d3a-4194-9264-a320a8d656ad

### CODE REVIEW

File: packages/observe/src/features/executions/components/ExecutionsListTable.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ export function ExecutionsListTable({
                             <Typography variant='body2'>{formatDate(execution.updatedDate)}</Typography>
                         </TableCell>
                         <TableCell>
-                            <Typography variant='body2'>{execution.agentflow?.name ?? '—'}</Typography>
+                            <Typography variant='body2'>{execution.agentflow?.name || '—'}</Typography>
```
[the remark this answers is not in the document]
Comment (jocelynlin-wd) [reply packages/observe/src/features/executions/components/ExecutionsListTable.tsx:92]: the change was from a previous PR comment https://github.com/FlowiseAI/Flowise/pull/6348#discussion_r3191704272

File: packages/observe/src/features/executions/components/ExecutionsViewer.tsx [hunk 2]
```diff
@@ -105,7 +106,7 @@ export function ExecutionsViewer({
         } finally {
             setIsLoading(false)
         }
-    }, [api, page, pageSize, agentflowId, filters])
+    }, [api, page, pageSize, agentflowIds, filters])
```
[the remark this answers is not in the document]
Comment (jocelynlin-wd) [reply packages/observe/src/features/executions/components/ExecutionsViewer.tsx]: Fixed — derived a stable agentflowIdsKey (joined string) and put it in the dep array; the callback re-derives the array from the key inside, so unmemoized parents passing fresh literals no longer cause re-fetches. Avoids the JSON.stringify-in-deps anti-pattern.

File: packages/server/src/controllers/executions/index.ts [hunk 3]
```diff
@@ -45,8 +45,19 @@ const getAllExecutions = async (req: Request, res: Response, next: NextFunction)
...
+        // mix `?agentflowId=a,b&agentflowId=c`. All flatten into a clean string[].
+        if (req.query.agentflowId !== undefined) {
+            const raw = req.query.agentflowId
+            const values = Array.isArray(raw) ? raw : [raw]
+            const ids = values
+                .flatMap((v) => String(v).split(','))
+                .map((v) => v.trim())
+                .filter(Boolean)
+            if (ids.length > 0) filters.agentflowIds = ids
+        }
```
[the remark this answers is not in the document]
Comment (jocelynlin-wd) [reply packages/server/src/controllers/executions/index.ts]: Fixed — switched to != null, and replaced String(v).split(',') with a typeof v === 'string' guard so qs's nested-object form (?agentflowId[x]=y) is dropped instead of producing '[object Object]'. Added two tests covering object-only and mixed-array inputs.
